src/components/form-widgets/price-field.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component, Fragment } from 'react'
+import { getFiatPrice } from 'utils/priceUtils'
+
+class PriceField extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      price: props.formData || '',
+      priceUsd: '0.00',
+      currencyCode: props.currencyCode || 'USD'
+    }
+
+    const { options } = props
+    const { selectedSchema } = options
+    const enumeratedPrice = selectedSchema && selectedSchema.properties['price'].enum
+    this.priceHidden =
+      enumeratedPrice &&
+      enumeratedPrice.length === 1 &&
+      enumeratedPrice[0] === 0
+  }
+
+  async componentDidMount() {
+    const { price, currencyCode } = this.state
+    if (this.props.formData) {
+      const priceUsd = await getFiatPrice(price, currencyCode)
+      this.setState({
+        priceUsd
+      })
+    }
+  }
+
+  onChange() {
+    return async (event) => {
+      const value = parseFloat(event.target.value)
+      const priceUsd = await getFiatPrice(value, this.state.currencyCode)
+      this.setState({
+        price: value,
+        priceUsd
+      }, () => this.props.onChange(value))
+    }
+  }
+
+  render() {
+    return(
+      this.priceHidden ?
+        <div></div> :
```

Comment:
Any reason to return this empty div rather than `null`?

Reply:
Good point! Pushing a change to `token-ui` branch